app/src/ui/dialog/is-top-most.tsx

```diff
@@ -0,0 +1,14 @@
+import memoizeOne from 'memoize-one'
+
+export function isTopMostDialog(
```

Comment:
This feels so much better, thanks for taking another swing at it ✨ Do you think you could add just a line or two of documentation for future us?